src/storage/src/source/generator.rs

```diff
@@ -239,6 +238,20 @@ fn render_simple_generator<G: Scope<Timestamp = MzOffset>>(
     let mut builder = AsyncOperatorBuilder::new(config.name.clone(), scope.clone());
 
     let (data_output, stream) = builder.new_output::<AccountedStackBuilder<_>>();
+    let partition_count = u64::cast_from(config.source_exports.len());
+    let data_streams: Vec<_> = stream.partition_core(
+        partition_count,
+        |((output, data), time, diff): &(
+            (usize, Result<SourceMessage, DataflowError>),
+            MzOffset,
+            Diff,
+        )| { (*output, (data.clone(), time.clone(), diff.clone())) },
+    );
+    let mut data_collections = BTreeMap::new();
+    for (id, data_stream) in config.source_exports.keys().zip_eq(data_streams) {
+        data_collections.insert(*id, data_stream.as_collection());
+    }
```

Comment:
This is another common piece of code that this PR adds to all implementations. This piece of code converts the legacy multiplexed output of sources into the non-multiplexed one that the new interface requires. These will be removed as we start moving individual sources into the new interface